lib/chef/provider/package/windows.rb

```diff
@@ -129,6 +129,7 @@ def action_install
         # Chef::Provider::Package action_install + action_remove call install_package + remove_package
         # Pass those calls to the correct sub-provider
         def install_package(name, version)
+          fail Chef::Exceptions::NoWindowsPackageSource, "installer_type '#{installer_type}' requires a source" if new_resource.source.nil?
```

Comment:
We do in fact require a source for install, the catch is that in the cookbook you could provide a source url as name, and it would work, but wouldn't be idempotent.

The meaningful "fix" here is probably in the cookbook. We change the cookbook to evaluate the `name` property to see if it is a URI, and if it is, move it to `source` before we give them the core provider but WARN that they need to fix that with a nice friendly message about how it wouldn't have worked.

what about this:

```
def define_resource_requirements
  requirements.assert(:install) do |a|
    a.assertion { @new_resource.source unless @package_provider == :msi }
    a.failure_message Chef::Exceptions::Package, "Source for package #{@new_resource.name} must be specified in the resource's source property for package to be installed because the package_name property is used to test for package installation state for this package type."
  end
end
```